casa/IO/test/tTypeIO.cc

```diff
@@ -131,34 +131,35 @@ void doIt (TypeIO* io)
 
 int main()
 {
-    RegularFileIO regularFileIO (Path("tTypeIO_tmp.dat"), ByteIO::New);
-    CanonicalIO canonicalIO (&regularFileIO);
+    std::shared_ptr<ByteIO> regularFileIO
+      (new RegularFileIO (Path("tTypeIO_tmp.dat"), ByteIO::New));
+    CanonicalIO canonicalIO (regularFileIO);
     doIt (&canonicalIO);
 
-    LECanonicalIO lecanonicalIO (&regularFileIO);
+    LECanonicalIO lecanonicalIO (regularFileIO);
     doIt (&lecanonicalIO);
 
-    RawIO rawIO (&regularFileIO);
+    RawIO rawIO (regularFileIO);
     doIt (&rawIO);
 
-    CanonicalDataConversion canConv;
-    ConversionIO canConvIO (&canConv, &regularFileIO);
+    std::shared_ptr<DataConversion> canConv (new CanonicalDataConversion());
```

Comment:
```suggestion
    auto canConv = std::make_shared<CanonicalDataConversion>();
```
Similar suggestions apply to the *canConv variables below.